packages/astro/src/@types/astro.ts

```diff
@@ -1375,6 +1375,72 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		devOverlay?: boolean;
+
+		// TODO review with docs team before merging to `main`
+		/**
+		 * @docs
+		 * @name experimental.i18n
+		 * @type {object}
+		 * @version 3.*.*
+		 * @type {object}
+		 * @description
+		 *
+		 * Allows to configure the beaviour of the i18n routing
+		 */
+		i18n?: {
+			/**
+			 * @docs
+			 * @name experimental.i18n.defaultLocale
+			 * @type {string}
+			 * @version 3.*.*
+			 * @description
+			 *
+			 * The default locale of your website/application
+			 */
+			defaultLocale: string;
+			/**
+			 * @docs
+			 * @name experimental.i18n.locales
+			 * @type {string[]}
+			 * @version 3.*.*
+			 * @description
+			 *
+			 * A list of locales supported by the website.
+			 */
+			locales: string[];
+
+			/**
+			 * @docs
+			 * @name experimental.i18n.fallback
+			 * @type {Record<string, string[]>}
+			 * @version 3.*.*
+			 * @description
+			 *
+			 * The fallback system of the locales. By default, the fallback system affect the **content only**, and it doesn't
+			 * do any redirects.
+			 *
+			 * This means that when attempting to navigate to a page that hasn't been translated, Astro will pull the content
+			 * from the page of the default locale and render it. No redirects will happen.
+			 */
+			fallback?: Record<string, string[]>;
+
+            /**
+             * @docs
+             * @name experimental.i18n.routingStrategy
+             * @type {'prefix-always' | 'prefix-other-locales'}
+             * @default {'prefix-other-locales'}
+             * @version 3.*.*
+             * @description
+             *
+             * Controls the routing strategy:
+             *  - `prefix-other-locales`: This is the default value. When used, Astro will identify those routes that belong to the default locale and execute a redirect to a route stripped of that locale.
+             *		Use `example.com/[lang]/content/` for every route.
+             *  - `prefix-always`: When used, all URLs need to contains a locale prefix. Astro will return a 404 for any route that doesn't fullfill the requirements.
+             * 		Use `example.com/[lang]/content/` for all non-default languages, but `example.com/content` for the default locale/.
+             *
```

Comment:
```suggestion
             * Controls the routing strategy to determine your site URLs.
             *
             *  - `prefix-other-locales` (default): Only non-default languages will display a language prefix. The `defaultLocale` will not show a language prefix.
             * URLs will be of the form `example.com/[lang]/content/` for all non-default languages, but `example.com/content/` for the default locale.
     
             *  - `prefix-always`: All URLs will display a language prefix.
             *  URLs will be of the form`example.com/[lang]/content/` for every route, including the default language.
             *
             * Note: Astro requires all content to exist within a `/[lang]/` folder, even for the default language. When configured to `prefix-other-locales`, Astro will execute a redirect to a route stripped of that locale.
```
This is significantly different, and I don't even think it's necessarily correct, but this is the kind of thing we have to be able to write, so I'll need you to fix this up in this format.  😅 

It's important here in this config setting that people are choosing based on what URLs they are going to get, because that's what they can see and that's what they know. They are going to choose this strategy not based on things like redirects behind the scenes, but based on the URL patterns they want.

So, I've reconstructed these to show which URL patterns you'll get if you choose each option. I've put the "how it works under the hood" separately below the options (and, I'm not even sure this is totally correct, so please correct!). This might not even necessary information to have here, really, as it can go in longer-form docs where we can explain how they need to structure their content. This is OK even if it just says when you configure X, you get these URLs. When you configure Y, you get these URLs.

Reply:
Like your format, just one thing that changed recently: https://github.com/withastro/roadmap/pull/734/commits/989ecc1c40eb4ad43acea78be3212ea3f688f974

Astro won't do any redirects in `prefix-always`, so URLs like `example.com/content` will result in a 404. I let you rephrase this.

Reply:
Sounds good! I'll edit the above suggestion to include that information, then you can fix whatever needs fixing!

Reply:
Suggestion updated! You can now work from that!